platform/engineering/design-docs/2020-08-17-mobile-api-design-doc.md

```diff
@@ -0,0 +1,173 @@
+
+### High Level Design
+Since vets-api already contains integrations to backend services that the mobile app will need 
+(such as MVI, Claims, Appointments), the mobile API will be built inside of the vets-api codebase, and exposed in the same 
+ way that VA.gov API endpoints are exposed.
+
+The mobile API will provide functionality that is currently exclusive to VA.gov as well as some already-public functionality 
+from Lighthouse, so existing integrations will be reused. A Rails engine will house the mobile API surface area, with necessary 
+functionality connected as closely as possible to the backend integrations. In some cases, this may be a simple passthrough, 
+while in other cases the mobile API will provide its own presentation layer on top of the 
+integrations. This will allow the mobile API to grow independently of the rest of vets-api, while providing a consistent API 
+that will be useful for future mobile applications to consume.
+
+All endpoints will be versioned, so that long-lasting mobile releases will continue to function into the future. When 
+modifications are required, changes will be made in a way to not break existing clients. The team is considering 
+building a machine-readable discovery API endpoint to decouple the mobile app from hard-coded urls.
+
+All mobile authorization will be confined to this Rails engine. The mobile app will require some new endpoints as well, and 
+these will be built in the Rails engine for isolation. [A companion auth design doc has more details](https://github.com/department-of-veterans-affairs/va.gov-team/pull/12482).
+
+## Specifics
+
+### Detailed Design
+The mobile endpoints need custom authentication, merged responses, and reduced payloads. With versioning this is possible 
+in the core api albeit with a fractured experience. Mobile endpoints could be on v0, v1, or even v2. MVP mobile endpoints 
+will be in flux, and parallel va.gov changes could cause versions to climb.
+
+To limit the disruption to the API it's best to perform the optimizations in a module with a `/mobile` base path. 
+This allows us to:
+
+- Add mobile authentication without changes to the core API codebase.
+- Stay out of the way as we tweak endpoints in a mobile sandbox.
+- Test out optimizations that can be later ported back to the core API.
+
+Expanding on the last point it's possible that once the mobile optimized versions of the endpoints have stabilized that we permanently
+move them back to core API. The aim is to avoid confusion about why a pereceived newer version of an endpoint would be missing fields
+without context that it was actually a 'mobile only' endpoint.
+
+#### Application Controller
+
+Authentication for the mobile app is via IAM's SSOe OAuth service. It is 100% API based and does not use callbacks or cookies. 
+The mobile `ApplicationController` would extend the core `ApplicationController` and override its authentication methods.
```

Comment:
maybe rename to something with a different identifier? (`MobileApplicationController`?)